sim/team-validator.js

```diff
@@ -493,9 +493,32 @@ class Validator {
 				if (eventProblems) problems.push(...eventProblems);
 			}
 		}
-		if (ruleTable.has('-illegal') && set.level < (template.evoLevel || 0)) {
-			// FIXME: Event pokemon given at a level under what it normally can be attained at gives a false positive
-			problems.push(`${name} must be at least level ${template.evoLevel} to be evolved.`);
+		// Minimum level validation
+		if (ruleTable.has('-illegal')) {
+			let minLevel = 0;
+			let allowHiddenAbility = [true, 0];
+			if (template.evoLevel) minLevel = template.evoLevel;
```

Comment:
Check `evoType.startsWith('level')` - if true, `minLevel` should be 1 + its prevo's `minLevel` (defaulting to 1).